### PR TITLE
[WIP] Add more Dummy cases

### PR DESCRIPTION
**Pull request by NicolasNSSM**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@
 src/**/*.css
 
 # misc
+.DS_Store
```
Comment (nicolasmure) [inline .gitignore:16]: this should be configured in global gitignore IMHO. I think you can remove this line (and tell mac users to configure their git :p )

File: .env.dist [hunk 2]
```diff
@@ -0,0 +1 @@
+REACT_APP_DEBUG_STATE=true
```
Comment (NicolasNSSM) [inline .env.dist]: Shouldn't we just set it as "false" in a versioned .env if we use .env.local?

File: src/Epic/Dog.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { combineEpics, ofType } from 'redux-observable'
+import * as Dog from '../Redux/State/Dog'
+import { mergeMap, map } from 'rxjs/operators'
+import { invoker } from 'ramda'
+
+export const fetchDogEpic = (action$, state$, { fetchDog }) => action$.pipe(
+  ofType(Dog.GENERATE_DOG),
+  mergeMap(fetchDog),
+  mergeMap(invoker(0, 'json')),
```
Comment (nicolasmure) [inline src/Epic/Dog.js:9]: why a mergeMap here ?
```javascript
map(response => response.json())
```
should be enough IMHO

**Review by nicolasmure — COMMENTED** [review]
Who let the dogs out ?

File: package.json [hunk 4]
```diff
@@ -26,5 +26,8 @@
     "build": "npm-run-all build-css build-js build-autoprefixed-css",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject"
+  },
+  "devDependencies": {
+    "react-test-renderer": "^16.4.1"
```
Comment (NicolasNSSM) [inline package.json:31]: Should we move "node-sass-chokidar" and "autoprefixer" there too?

File: src/Redux/State/Dog.spec.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import * as Dog from './Dog'
+
+describe('Redux :: Module :: Dog', () => {
+  it('reduce to initial state', () => {
+    const state = Dog.default()
+
+    expect(state).toEqual(Dog.INITIAL_STATE);
+  });
+
+  it('reduce the generate dog action', () => {
```
Comment (NicolasNSSM) [inline src/Redux/State/Dog.spec.js]: s/reduce/reduces/

File: src/Redux/State/Dog.spec.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    const s1     = Dog.default();
+    const action = Dog.generateDog();
+    const s2     = Dog.default(s1, action);
+
+    expect(s2).not.toEqual(s1)
+
+    expect(s2.isGenerating).toBe(true);
+  });
+
+  it('reduce receive dig action', () => {
```
Comment (NicolasNSSM) [inline src/Redux/State/Dog.spec.js]: s/reduce/reduces/

File: src/Redux/State/index.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import Dog from './Dog'
+import { tap, pipe } from 'ramda'
+import { combineReducers } from 'redux'
+
+// debug :: ((State, Action *) -> State) -> -> State -> Action * -> State
```
Comment (NicolasNSSM) [inline src/Redux/State/index.js]: Somehting wrong here with `-> ->`

File: src/Redux/State/index.spec.js [hunk 8]
```diff
@@ -1,9 +1,9 @@
 import * as Module from './index'
 
 describe('Redux :: Module', () => {
-  it('it contains the Dummy state', () => {
+  it('it contains the Dog state', () => {
```
Comment (NicolasNSSM) [inline src/Redux/State/index.spec.js]: it it !

File: .gitignore (hunk 1 above)
Comment (NicolasNSSM) [reply .gitignore:16]: It's added by react-create-app by default, I removed it then we decided it cost nothing to leave it there, but I agree with you, this has nothing to do here IMO

File: src/index.js [hunk 9]
```diff
@@ -1,9 +1,27 @@
...
+import { applyMiddleware } from 'redux'
+import { createEpicMiddleware } from 'redux-observable'
+import { Store } from './Redux/Store'
+import rootEpic from './Epic'
+import { default as mainReducer, debug } from './Redux/State'
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const epicMiddleware = createEpicMiddleware({
+  dependencies: {
+    fetchDog: () => fetch('https://dog.ceo/api/breeds/image/random')
```
Comment (NicolasNSSM) [inline src/index.js]: missing trailing comma

File: src/Epic/Dog.js (hunk 3 above)
Comment (NicolasNSSM) [reply src/Epic/Dog.js:9]: AFAIR The fetch api returns a promise of a response so you have to merge'em all before using the response

**Pull request merged**